monai/metrics/hausdorff_distance.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union
+
+import numpy as np
+import torch
+from scipy.ndimage.morphology import binary_dilation
```

Comment:
the import from optional packages should be done with:
```py
from monai.utils import optional_import
binary_dilation, _ = optional_import("scipy.ndimage.morphology", name="binary_dilation")
directed_hausdorff, _ = optional_import("scipy.spatial.distance", name="directed_hausdorff")
```
so that the users who don't have `scipy` and don't use this metric can still use the other MONAI features

Reply:
Thanks!